### ISSUE TITLE
unbound variable check on R6RS is done incorrectly.

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#185]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/180) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This should not raise an error.
```
#!scheme
(library (helper)
    (export mark)
    (import (rnrs))
(define mark #'mark)
)
(library (bug)
    (export foo)
    (import (rnrs) (for (helper) expand))

(define-syntax foo
  (lambda (x)
    (display mark) (newline)
    (syntax-case x ()
      ((_) 'ok))))
)
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This happens only `-r6` option is passed and the reported script is written in one file. So it's more like `program` hidden syntax's problem.

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Executing libraries inside of 'program' syntax. (Fixes #180)
